test/unit/event_store/repository_test.rb

```diff
@@ -28,7 +30,7 @@ def test_publish_event
   end
 
   def test_active_record_store
-    store = Repository.new(RailsEventStoreActiveRecord::EventRepository.new)
+    store = Repository.new(RailsEventStoreActiveRecord::EventRepository.new(adapter: EventStore::Event))
```

Comment:
Took me a while to figure this one out... 

`EventStore::Event` implements serialization using GlobalId, where the objects are serialized by transforming into a global ID looking like `aj_globalid: gid://system/Account/28` (and later deserialized by fetching the object from the DB by ID).

Without this, the test was trying to serialize and deserialize the object Account (`provider` field in the event data), and this was failing with:

```
Tried to load unspecified class: Account
```

All the real events should be going through the repository with the correct adapter, so this is only a tests issue.

Reply:
nice investigation!